fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -1492,6 +1494,48 @@ private List<ResultsetRowData> fillDatatableResultSetDataRows(final String sql)
         return resultsetDataRows;
     }
 
+    private List<ResultsetRowData> fillDatatableResultSetDataRows(final String sql, final List<ResultsetColumnHeaderData> columnHeaders) {
+        final List<ResultsetRowData> resultsetDataRows = new ArrayList<>();
+        final GenericResultsetData genericResultsetData = new GenericResultsetData(columnHeaders, null);
+
+        final SqlRowSet rowSet = jdbcTemplate.queryForRowSet(sql); // NOSONAR
+        final SqlRowSetMetaData rsmd = rowSet.getMetaData();
```

Comment:
Isn't this whole logic pretty much a duplicate of the fillDatatableResultSetDataRows#String version? We should definitely extract whatever is common instead of duping it completely.

Reply:
Done!